### PR TITLE
Code: Add staticfiles to urlpatterns in vms/urls.py

### LINKED ISSUE #951: fix serving uploaded docs through media
## Description
As a Member,
I need to add static files path to URL patterns in the vms urls.py,
so that I can serve the uploaded documents.

## Mocks
image

## Steps to reproduce
To serve the uploaded media files through URL like 
srv/vms/resume/[pdf_name] 
this URL should be able to view the media files

### Update [Required]
- [ ] vms urls.py file 
- [ ] .gitignore

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
10 minutes

### PR DESCRIPTION
**Pull request by sankha555**
# Description
This pull request adds staticfiles and media root to urlpatterns in urls.py file of vms module. 

Fixes #951

# Type of Change:
**Delete irrelevant options.**

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
The code has been run on my local computer.

# Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

**Screenshots**
Browser output after signing up - no errors
image

Terminal Logs of running server - no errors and email also confirmed
image

### CODE REVIEW

**Comment by yashshah2820** [comment]
@sankha555 tick mark the point in the checklist

File: .gitignore [hunk 1]
```diff
@@ -103,3 +103,7 @@ local/
 # Documentation build
 _build
 
+# static files generated from Django application using `collectstatic`
+media
+static
```
Comment (yashshah2820) [inline .gitignore:108]: Why these changes, only add changes related to the issue, please remove these.

**Comment by sankha555** [comment]
@yashshah2820 , I have made the changes requested by you in the PR. Kindly review my changes. Thanks!

File: vms/volunteer/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                ('resume_file', models.FileField(blank=True, max_length=75, upload_to='vms/resume/')),
+                ('reminder_days', models.IntegerField(blank=True, default=1, validators=[django.core.validators.MaxValueValidator(50), django.core.validators.MinValueValidator(1)])),
+                ('city', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='cities_light.City')),
+                ('country', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='cities_light.Country')),
+                ('organization', models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, to='organization.Organization')),
+                ('state', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='cities_light.Region')),
+                ('user', models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
+            ],
+        ),
+    ]
```
Comment (yashshah2820) [inline vms/volunteer/migrations/0001_initial.py:43]: remove this file it is not needed

File: vms/shift/migrations/0002_auto_20200510_1926.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            name='volunteer',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='volunteer.Volunteer'),
+        ),
+        migrations.AddField(
+            model_name='shift',
+            name='city',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='cities_light.City'),
+        ),
+        migrations.AddField(
+            model_name='shift',
```
Comment (yashshah2820) [inline vms/shift/migrations/0002_auto_20200510_1926.py:32]: remove this file

File: vms/shift/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('start_time', models.TimeField(blank=True, null=True)),
+                ('end_time', models.TimeField(blank=True, null=True)),
+                ('date_logged', models.DateTimeField(blank=True, null=True)),
+                ('edit_requested', models.BooleanField(default=False)),
+                ('report_status', models.BooleanField(choices=[(False, 'Not reported'), (True, 'Reported')], default=False)),
+                ('shift', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='shift.Shift')),
+            ],
+        ),
```
Comment (yashshah2820) [inline vms/shift/migrations/0001_initial.py:59]: remove this file

File: vms/organization/migrations/0001_initial.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    operations = [
+        migrations.CreateModel(
+            name='Organization',
+            fields=[
+                ('id', models.AutoField(primary_key=True, serialize=False)),
+                ('name', models.CharField(max_length=75, unique=True, validators=[django.core.validators.RegexValidator("^[(A-Z)|(a-z)|(0-9)|(\\s)|(\\-)|(:)|(\\')]+$")])),
+                ('approved_status', models.IntegerField(default=1)),
+            ],
+        ),
+    ]
```
Comment (yashshah2820) [inline vms/organization/migrations/0001_initial.py:25]: remove this file

File: vms/job/migrations/0001_initial.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            fields=[
+                ('id', models.AutoField(primary_key=True, serialize=False)),
+                ('name', models.CharField(max_length=75, validators=[django.core.validators.RegexValidator("^[(A-Z)|(a-z)|(\\s)|(\\')]+$")])),
+                ('start_date', models.DateField()),
+                ('end_date', models.DateField()),
+                ('description', models.TextField(blank=True, validators=[django.core.validators.RegexValidator("^[(A-Z)|(a-z)|(0-9)|(\\s)|(\\.)|(,)|(\\-)|(!)|(\\')]+$")])),
+                ('event', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='event.Event')),
+            ],
+        ),
+    ]
```
Comment (yashshah2820) [inline vms/job/migrations/0001_initial.py:30]: remove this file please

File: vms/event/migrations/0001_initial.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                ('start_date', models.DateField()),
+                ('end_date', models.DateField()),
+                ('address', models.CharField(blank=True, max_length=75, null=True, validators=[django.core.validators.RegexValidator("^[(A-Z)|(a-z)|(0-9)|(\\s)|(\\-)|(\\')|(,)]+$")])),
+                ('venue', models.CharField(blank=True, max_length=30, null=True, validators=[django.core.validators.RegexValidator("^[(A-Z)|(a-z)|(\\s)|(\\-)|(\\')]+$")])),
+                ('city', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='cities_light.City')),
+                ('country', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='cities_light.Country')),
+                ('state', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='cities_light.Region')),
+            ],
+        ),
+    ]
```
Comment (yashshah2820) [inline vms/event/migrations/0001_initial.py:34]: remove this file, there is no need for migration files

File: vms/administrator/migrations/0001_initial.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        ('cities_light', '0008_city_timezone'),
+        ('organization', '0001_initial'),
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Administrator',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
```
Comment (yashshah2820) [inline vms/administrator/migrations/0001_initial.py:25]: remove this file

**Comment by sankha555** [comment]
@yashshah2820 , as requested by you, I have deleted all the files that weren't required. Kindly review my PR. Also, I wanted to know that for future contributions, which method should I preferably use for not committing the migration files : should I straightaway delete the files from my system and start afresh every time or selectively delete the files after a commit? Kindly guide me in this regard. Thanks!

**Comment by yashshah2820** [comment]
you can add the migration folder to gitignore and while committing to git, do not stage gitignore. Hope it helps 🙂

**Comment by yashshah2820** [comment]
Well done @sankha555 👍

**Comment by yashshah2820** [comment]
@anitab-org/qa-team this PR is ready to be merged to develop

**Comment by yashshah2820** [comment]
@sankha555 try to squash the commits into one

**Comment by sankha555** [comment]
Done the squashing @yashshah2820. Thanks for your guidance and motivation :) !

**Comment by rpattath** [comment]
@sankha555 @yashshah2820 Please provide steps to reproduce the scenario in issue. I saw this [screenshot](https://user-images.githubusercontent.com/42682370/76887015-9c571000-68a7-11ea-863b-bf03e6c6f980.png) on the issue but could you provide the steps so that I can make sure that issue is not seen with the fix in this PR.

**Comment by yashshah2820** [comment]
> @sankha555 @yashshah2820 Please provide steps to reproduce the scenario in issue. I saw this [screenshot](https://user-images.githubusercontent.com/42682370/76887015-9c571000-68a7-11ea-863b-bf03e6c6f980.png) on the issue but could you provide the steps so that I can make sure that issue is not seen with the fix in this PR.

I didn't understand what are you asking for?

**Comment by yashshah2820** [comment]
@rpattath please respond to the comments

**Comment by rpattath** [comment]
> > @sankha555 @yashshah2820 Please provide steps to reproduce the scenario in issue. I saw this [screenshot](https://user-images.githubusercontent.com/42682370/76887015-9c571000-68a7-11ea-863b-bf03e6c6f980.png) on the issue but could you provide the steps so that I can make sure that issue is not seen with the fix in this PR.
> 
> I didn't understand what are you asking for?

Sorry for the delay in response. I need steps to reproduce the issue that this PR fixes which will help me to test the PR. I saw a screenshot in this issue https://github.com/anitab-org/vms/issues/951 which I am not sure how to get to. Hope this clarifies your question.

**Comment by yashshah2820** [comment]
@rpattath I have added some reproducing steps hope it helps :)

**Comment by rpattath** [comment]
@sankha555 @yashshah2820 When I trying to run the project from the branch with this PR I hit the issue in the following screen shot. I was trying to sign-up as a user.

Screenshot from 2020-05-16 16-30-02

**Comment by Harirai** [comment]
@rpattath That error occurs when you fill an organization which is already available in drop down. (Already present in the database).
That should be a separate issue.

**Comment by rpattath** [comment]
@sankha555 @yashshah2820 I was able to test the fix successfully for a Resume in pdf format. When I uploaded a docx format file, I see that the doc is being downloaded instead of viewed. Is that the expected behavior? Should we restrict to uploading only pdf files?

**Comment by yashshah2820** [comment]
> @sankha555 @yashshah2820 I was able to test the fix successfully for a Resume in pdf format. When I uploaded a docx format file, I see that the doc is being downloaded instead of viewed. Is that the expected behavior? Should we restrict to uploading only pdf files?

If any other feature is required we have to make a new issue and I was able to view the resume from the admin panel this PR is ready to be merged

File: .gitignore [hunk 9]
```diff
@@ -103,3 +103,4 @@ local/
 # Documentation build
 _build
 
+
```
Comment (Monal5031) [inline .gitignore:106]: 🤔 extra newline?

Comment (yashshah2820) [reply .gitignore:106]: @sankha555 see to it :)

**Comment by sankha555** [comment]
I guess that happenee when adding some lines to the gitignore file and then removing them as requested by @yashshah2820 . I supose that extra new line must have crept in then 😅

File: .gitignore (hunk 9 above)
Comment (sankha555) [reply .gitignore:106]: I guess that happenee when adding some lines to the gitignore file and then removing them as requested by @yashshah2820 . I supose that extra new line must have crept in then 😅

Comment (yashshah2820) [reply .gitignore:106]: please remove the extra line from the file

Comment (sankha555) [reply .gitignore]: @yashshah2820 Done! The extraline has been removed.
